src/utilities/input.tsx

```diff
@@ -137,24 +145,35 @@ export const InputActions = ({
 
 InputActions.displayName = 'FondueInputActionButtons';
 
-export const InputExtraActions = ({ actions, disabled, dataTestId }: InputExtraActionsProps) => {
+export const InputExtraActions = ({
+    actions,
+    disabled,
+    dataTestId,
+}: InputExtraActionsProps): (ReactElement | null)[] => {
     return actions.map((action) => {
+        const { element, tooltip } = action;
         const id = generateRandomId();
-        return (
-            <Button
-                key={`extra-action-${id}`}
-                style={ButtonStyle.Default}
-                onClick={action.callback}
-                disabled={disabled}
-                emphasis={ButtonEmphasis.Weak}
-                icon={action.icon}
-                size={ButtonSize.Small}
-                aria-describedby="Clear Input Value"
-                aria-label="Clear value"
-                title={action.title}
-                data-test-id={`${dataTestId}-extra-action`}
-            />
-        );
+        if (tooltip) {
+            return (
+                <Tooltip
+                    content={tooltip.content}
+                    key={`extra-action-tooltip-${id}`}
+                    disabled={tooltip.disabled ?? disabled}
+                    data-test-id={`${dataTestId}-extra-action`}
+                >
+                    {cloneElement(tooltip.children, { disabled: tooltip.disabled ?? disabled })}
```

Comment:
Is it the reason why we choose between Tooltip and Element? Like the whole Tooltip doesn't work if passed as ReactElement?

Reply:
@imoutaharik I am update this to allow any `ReactElement`.